src/storage/engine/mod.rs

```diff
@@ -359,6 +364,8 @@ impl<I: Iterator> Cursor<I> {
     /// This method assume the current position of cursor is
     /// around `key`, otherwise you should use `seek` instead.
     pub fn near_seek(&mut self, key: &Key, statistics: &mut CFStatistics) -> Result<bool> {
+        self.valid = true;
+
```

Comment:
Also should reset valid in L371.

Reply:
For the same reason as above, this is a real iterator seek, so we don't need to mark the flag.